### DISCUSSION TITLE
Why WebViewManager can still send messages to WebView2 after BlazorWebView Unloaded event.

### DISCUSSION THREAD

**Discussion by yurastanchuk** [MicrosoftEdge/WebView2Feedback#4044] [Q&A]
Because of this [issue ](https://github.com/MicrosoftEdge/WebView2Feedback/issues/1424) , I have to dispose webview processes manually. I do it as following after window close event: 
<img width="395" alt="image" src="https://user-images.githubusercontent.com/19996151/235806836-3664bfa7-2ce6-4144-b85b-5dce691aaaf4.png">

But still even with delay of 10 seconds after BlazorWebView Unloaded event, I still sometimes get error 'Cannot access a disposed object'.  Why this can happen and how to properly close  webview process to avoid this error? 

Full error: [error.txt](https://github.com/MicrosoftEdge/WebView2Feedback/files/11378347/error.4.txt)

**Comment by yurastanchuk** [comment]
upd: the message received after BlazorWebView Unloaded event is : "__bwv:[\"OnRenderCompleted\",6,null]"

**Comment by MackinnonBuck** [comment]
@yurastanchuk, it looks like the problem is that the code in your screenshot is directly disposing the underlying `WebView2Control` rather than disposing the `BlazorWebView` itself. This will result in other resources managed by `BlazorWebView` to continue attempting to access the underlying `WebView2Control` without knowing it has been disposed. Could you please try invoking `webView.DisposeAsync()` instead? Thanks!

**Comment by yurastanchuk** [comment]
The same error after i have changed the code to : 
<img width="423" alt="image" src="https://github.com/MicrosoftEdge/WebView2Feedback/assets/19996151/265240ff-a0c3-4343-81fe-443a3a6e61fd">

**Comment by MackinnonBuck** [comment]
@yurastanchuk could you please provide a minimal repro project demonstrating the issue, hosted as a public GitHub repository? This will help us investigate the issue further. Thanks!
